utils/index.js

```diff
@@ -35,11 +36,7 @@ const getScore = (markedChoices, correctChoiceIds, possibleChoices) => {
   const correctlyAnswered = possibleChoices.filter(choice => R.contains(choice.id, correctlyAnsweredIds))
   const incorrectlyAnswered = possibleChoices.filter(choice => R.contains(choice.id, incorrectlyAnsweredIds))
 
-  let score = correctlyAnswered.reduce((acc, c) => acc + c.positiveWeight, 0)
-  score += incorrectlyAnswered.reduce((acc, c) => acc + c.negativeWeight, 0)
-
-  // scale score out of 10
-  score *= 10
+  const score = R.equals(new Set(correctChoiceIds), new Set(correctlyAnsweredIds)) ? question.positiveScore : -question.negativeScore
```

Comment:
R.equals is a performance killer, better avoid it if we can. It performs deep equality checks.

To check my hypothesis I wrote this benchmark test: https://jsben.ch/ZXF47 

if two sets(s1, s2) are equal, their union set is of the same size as that s1 and s2.

Reply:
Would like to add, in general Ramda is not optimized for performance (comparing to vanilla es6 features). Sure, they have thier use case as they allow better abstractions ans save dev time , but avoid them for trivial things.

Reply:
done